Src/PerformanceCollector/Shared/QuickPulseTelemetryModule.cs

```diff
@@ -0,0 +1,468 @@
+
+            if (this.telemetryProcessor == null)
+            {
+                QuickPulseEventSource.Log.CouldNotObtainQuickPulseTelemetryProcessorEvent();
+
+                throw new ArgumentException("Could not obtain an IQuickPulseTelemetryProcessor");
+            }
+
+            this.telemetryProcessor.Initialize(this.serviceClient.ServiceUri);
+        }
+
+        private void InitializePerformanceCollector()
+        {
+            foreach (var counter in QuickPulsePerfCounterList.CountersToCollect)
+            {
+                PerformanceCounter pc = null;
+                bool usesPlaceholder;
+
+                try
+                {
+                    pc = PerformanceCounterUtility.ParsePerformanceCounter(counter.Item2, null, null, out usesPlaceholder);
+                }
+                catch (Exception e)
+                {
+                    QuickPulseEventSource.Log.CounterParsingFailedEvent(e.Message, counter.Item2);
+                    continue;
+                }
+
+                if (usesPlaceholder)
+                {
+                    throw new InvalidOperationException(
+                        "Instance placeholders are not currently supported since they require refresh. Refresh is not implemented at this time.");
+                }
+
+                try
+                {
+                    this.performanceCollector.RegisterPerformanceCounter(
+                        counter.Item2,
+                        counter.Item1.ToString(),
+                        pc.CategoryName,
+                        pc.CounterName,
+                        pc.InstanceName,
+                        false,
+                        true);
+
+                    QuickPulseEventSource.Log.CounterRegisteredEvent(counter.Item2);
+                }
+                catch (InvalidOperationException e)
+                {
+                    QuickPulseEventSource.Log.CounterRegistrationFailedEvent(e.Message, counter.Item2);
+                }
+            }
+        }
+
+        private void ValidateConfiguration(TelemetryConfiguration configuration)
+        {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            if (configuration.TelemetryInitializers == null)
+            {
+                throw new ArgumentNullException(nameof(configuration.TelemetryInitializers));
+            }
+        }
+
+        private void InitializeTimers()
+        {
+            this.collectionTimer = new Timer(this.CollectionTimerCallback);
+
+            this.stateTimer = new Timer(this.StateTimerCallback);
+            this.stateTimer.ScheduleNextTick(this.initialDelay);
+        }
+
+        private IQuickPulseTelemetryProcessor FetchTelemetryProcessor(TelemetryConfiguration configuration)
+        {
+            return configuration.TelemetryProcessors.OfType<IQuickPulseTelemetryProcessor>().SingleOrDefault();
```

Comment:
Agreed offline that a module is still somewhat advantageous since it exists in a single instance across telemetry configurations.